Shopify/cli
# [Theme] Fix refreshing empty, deleted and new files

### WHY are these changes introduced?

Closes https://github.com/Shopify/develop-advanced-edits/issues/308
Closes https://github.com/Shopify/develop-advanced-edits/issues/301



### WHAT is this pull request doing?



### How to test your changes?



Each of the following steps should trigger a hot reload or a full refresh, depending on the information we have:

1. Make a CSS file empty and save => CSS is lost
1. Restore it => back to normal
1. Delete a CSS file => CSS is lost, and there's a 404 in network tab
1. Restore it => back to normal
1. Make a section file empty and save => section is gone
1. Restore it => back to normal
1. Delete a section file => section is gone
1. Restore it => back to normal
1. Delete any other file (template, JS asset) => full refresh

cc @Shopify/advanced-edits
